### ISSUE TITLE
JavaScript errors in IE7 

### ISSUE THREAD

**Issue by pgbakker** [mozilla/persona#831]
Hi, 

When I open my site that has BrowserID integrated in IE7, IE7 reports 2 errors, both of the same type: 
SCRIPT5022: Exception thrown and not caught 
include.js, line 37 character 2805
SCRIPT5022: Exception thrown and not caught 
production.js, line 37 character 1861

The "include.js" file I'm using comes from "https://browserid.org/include.js"

I have no issue that IE7 is not supported (I do get the unsupported browser dialog when trying to authenticate using BrowserID), but I rather not have the site report errors to the user.

As far as I know, the errors reported related to throwing exceptions from JavaScript that are not withing try/catch blocks.

Is there something that can be done to get rid of these errors?

I tested with the latest version of IE7 (7.0.5730.13) AFAIK.

Regards,

P.

**Comment by pgbakker** [comment]
Note: the issue can be reproduced in IE9 by switching to the Browser Mode to IE7 and the Document Mode to IE7 Standards using the Developer tools.

P.

**Comment by pgbakker** [comment]
Forgot to say that is can be reproduced on http://myfavoritebeer.org/ as well

**Comment by benadida** [comment]
@pgbakker we should absolutely address this. Thanks for bringing it to our attention.

**Comment by pgbakker** [comment]
Don't think there's any Windows version that, installed out of the box still bundles IE7.

However, we still see a large number of IE7 users hitting our website, according to Google Analytics, so I prefer to not have them see our website having errors (which in IE7 is always visible in the lower left corner).

P.

**Comment by pgbakker** [comment]
Interesting... Since you didn't get the same errors I got, I went back and retested our site now in both IE9 with IE7 mode simulation and in a VM with a real IE7.

And in both I cannot reproduce neither the errors I got before, but also not the error you're mentioning about localstorage.

I do get the dialog saying that the browser is not supported, but that's OK by me.

so, from my point of view, the errors I experienced have disappeared somehow and I don't get any other errors, so for me the issue has disappeared so the case can be closed.

P.
